client/js/socket-events/msg.js

```diff
@@ -47,4 +58,15 @@ socket.on("msg", function(data) {
 			.find(".unread-marker")
 			.appendTo(container);
 	}
+
+	if ((data.msg.type === "message" || data.msg.type === "action") && chan.hasClass("channel")) {
```

Comment:
I moved this autocompletion ordering here as there is no need to apply this logic when rendering bulk messages (e.g. when opening the page), as the server now keeps track of messages. I also removed `notice` from here as notices always target query windows and lobbies, and neither of these have user list.